sdks/python/apache_beam/testing/util.py

```diff
@@ -154,41 +154,50 @@ def equal_to_per_window(expected_window_to_elements):
 # Note that equal_to checks if expected and actual are permutations of each
 # other. However, only permutations of the top level are checked. Therefore
 # [1,2] and [2,1] are considered equal and [[1,2]] and [[2,1]] are not.
-def equal_to(expected):
+def equal_to(expected, equals_fn=None):
+  equal_to.equals_fn=equals_fn
```

Comment:
This creates a side-effect, making `equal_to` not thread-safe. Can we access `equals_fn` from the outer scope as we do with `expected` ?

Reply:
Yes, I tried that but it indeed does not work. PTAL and the alternative I just committed.